docs/mimxrt/machine.QECNT.rst

```diff
@@ -0,0 +1,377 @@
+        from the 0 up to cpc - 1, and then reset to the init value and increase
+        the cycles counter by one. The default is: no cpc set. In that case the
+        position counter overflows at 2**32 - 1. When counting down, the cycles counter changes
+        at the transition from 0 to cpc - 1.
+      - *compare*\=value. Sets a position counter compare value. When the counter matches
+        this value, a callback function can called and the match output will get a high level.
+      - *signed*\=False|True tells, whether the value return by Encoder.value() is signed or
+        unsigned. The default is ``True``.
+
+The arguments phase_a, phase_b and filter_ns are generic across ports, all other arguments are port-specific.
+
+Methods
+-------
+
+.. method:: Encoder.init(*, phase_a, phase_b, home, reverse, filter_ns, index, match, cpc, compare, signed)
+
+   Modify settings for the Encoder object.  See the above constructor for details
+   about the parameters.
+
+.. method:: Encoder.deinit()
+
+   Stops the Encoder, disables interrupts and releases the resources used by the encoder. On
+   Soft Reset, all instances of Encoder and Counter are deinitialized.
+
+.. method:: value=Encoder.value([value])
+
+   Get or set the current position counter of the Encoder as signed or unsigned 32 bit integer,
+   depending on the signed=xxx keyword option of init().
+
+   With no arguments the actual position counter value is returned.
+
+   With a single *value* argument the position counter is set to that value and the
+   cycles counter is cleared. The methods returns the previous value.
+
+.. method:: cycles=Encoder.cycles([value])
+
+   Get or set the current cycles counter of the Encoder as signed 16 bit integer.
+
+   With no arguments the actual cycles counter value is returned.
+
+   With a single *value* argument the cycles counter is set to that value. The
+   position counter is not changed. The methods returns the previous value.
+
+   If the value returned by Encoder.value() is unsigned,
+   the total position can be calculated as cycles() * cpc + value().
+   If the total position range is still too small, you can create your own cycles
+   counter using the irq() callback method.
+
+.. method:: Encoder.compare([value])
+
+   Get or set the position compare value.
+
+   With no arguments the actual compare value is returned. With a single *value* argument
+   the compare value is set to that value.
+
+.. method:: Encoder.irq(trigger=event, handler=handler, hard=False)
+
+   Specifies, that the *handler* is called when the respective *event* happens.
+
+   *event* may be:
+    - Encoder.COMPARE Triggered when the position counter matches the compare value.
+    - Encoder.ROLL_OVER Triggered when the position counter rolls over from the highest
+      to the lowest value.
+    - Encoder.ROLL_UNDER Triggered when the position counter rolls under from the lowest
+      to the highest value.
```

Comment:
It seems to me that it would be more correct to call
Encoder.IRQ_COMPARE
Encoder.IRQ_ROLL_OVER
Encoder.IRQ_ROLL_UNDER
like the
Pin.IRQ_FALLING
Pin.IRQ_RISING

Reply:
Being a lazy typer, I prefer shorter names.

Reply:
In the vast majority of cases, MicroPython uses IRQ_XXX.
https://docs.micropython.org/en/latest/search.html?q=irq&check_keywords=yes&area=default#